packages/smooth_app/lib/themes/smooth_theme.dart

```diff
@@ -122,8 +122,8 @@ class SmoothTheme {
       ),
       textTheme: _TEXT_THEME,
       floatingActionButtonTheme: FloatingActionButtonThemeData(
-        backgroundColor: myColorScheme.secondary,
-        foregroundColor: myColorScheme.onSecondary,
+        backgroundColor: myColorScheme.primary,
+        foregroundColor: myColorScheme.onPrimary,
```

Comment:
@bhattabhi013 Actually my idea was sightly different: assuming that we don't want to use a secondary color because we're very happy with only one color (called the primary), the idea was to set the secondary/onSecondary colors to the very same value as the primary/onPrimary colors.

This way you don't even have to set the floating action colors: the default values are secondary/onSecondary, which happen to be the same as primary/onPrimary.

And that would work for dark mode too, where we can decide to set the secondary/onSecondary to be different from primary/onPrimary, and therefore more visible, in dark mode only.

Besides, could you please get rid of `FloatingActionButton` explicit colors when it's primary or secondary in the whole app: unless in some rare cases we use the default color (= secondary) (= primary).

Reply:
Hi @monsieurtanuki,
I don't think it'll be a great idea to set the secondary color to the primary color's value because it'll be an extra color (secondary color) having the same value and it can cause confusion. 
I'll remove the explicit colors used in `FloatingActionButton` for the app.

Reply:
Hi @bhattabhi013!

The root of the confusion is using the same color for primary needs (e.g. the app bar) and secondary needs (e.g. the floating action button). But it's a choice I respect and that complies with Material Design.

Let me quote again https://material.io/design/color/the-color-system.html#color-theme-creation:
> A _secondary_ color provides more ways to accent and distinguish your product. Having a secondary color is _optional_, and should be applied sparingly to accent select parts of your UI.
> If you don’t have a secondary color, your primary color can also be used to accent elements.
> Secondary colors are best for: Floating action buttons, Selection controls, like sliders and switches, Highlighting selected text, Progress bars, Links and headlines

The thing is that apparently nobody wants the teal/black colors anywhere in the app, and that's the default color for secondary/onSecondary in flutter. Hence my suggestion to overwrite the default colors for secondary, with the primary color values.

I won't fight for that - though I know I'm probably right ;)

But the most important thing is to have cohesion in the app: no need to specify colors for FAB (unless in very specific cases), because we'll just use the default value. Will that default be the overwritten values for secondary/onSecondary (my suggestion) or the overwritten values for FABTheme (your suggestion)? Not very important after all, but you'll probably see the teal/black combination on some other `Widget`s because they rely on secondary/onSecondary, and you'll have to change the specific themes or individual colors `Widget` after `Widget`.

Reply:
Hi @monsieurtanuki,
Thanks for the detailed explanation. There was confusion as I had a notion that the color needs to be changed only for this particular widget(FAB) but after going through your comments again, I feel your solution is more suitable for the app.
I've changed the secondary to the primary color in the `smooth_theme`. Moreover, if we need any additional color we can always do that in the future, and hence no issues.